src/Components/Stories/SignUp/SignUp.jsx

```diff
@@ -1,36 +1,235 @@
+            setState({
+                ...inpState,
+                last_name: inputStore(name, '')
+            })
+        } else {
+            setState({
+                ...inpState,
+                last_name: inputStore(name, 'Please enter your correct Last name')
+            })
+        }
+    }
+    const emailChange = (e) => {
+        const email = e.target.value;
+        const emailPattern = /\w+([.-_]\w+)*@\w+([.-_]\w+)*\.\w{2,4}/;
+
+        if (emailPattern.test(email)) {
+            setState({
+                ...inpState,
+                mail: inputStore(email, '')
+            })
+        } else {
+            setState({
+                ...inpState,
+                mail: inputStore(email, 'Please enter valid Email id')
+            })
+        }
+    }
+    const phoneChange = (e) => {
+        const phone = e.target.value;
+        const phonePattern = /^\d{10}$/;
+
+        if (phonePattern.test(phone)) {
+            setState({
+                ...inpState,
+                phone: inputStore(phone, '')
+            })
+        } else {
+            setState({
+                ...inpState,
+                phone: inputStore(phone, 'Please enter valid mobile number')
+            })
+        }
+    }
+    const passwordChange = (e) => {
+        const pass = e.target.value;
+        const passPattern = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/;
+
+        if (passPattern.test(pass)) {
+            setState({
+                ...inpState,
+                password: inputStore(pass, ''),
+                re_password: inputStore('', '')
+            })
+        } else {
+            setState({
+                ...inpState,
+                password: inputStore(pass, 'Please enter Strong Password'),
+                re_password: inputStore('', '')
+            })
+        }
+    }
+    const passwordMatch = (e) => {
+        const rePass = e.target.value;
+
+        if (rePass === inpState.password.value) {
+            setState({
+                ...inpState,
+                re_password: inputStore(rePass, '')
+            })
+        } else {
+            setState({
+                ...inpState,
+                re_password: inputStore(rePass, 'Password does not match')
+            })
+        }
+    }
+    const signUp = async (e) => {
+        setState({
+            ...inpState,
+            signUpClicked: true
+        })
+        e.preventDefault()
+        const userPostData = {
+            last_name: last_name.value,
+            first_name: first_name.value.trim(),
+            mail: mail.value,
+            phone: phone.value,
+            password: password.value
+        }
+        const apiData = postSignUpUserApi(axios, userPostData);
+        toast.promise(apiData, {
+            pending: {
+                render() {
+                    return ('Please wait...')
+                },
+                position: toast.POSITION.TOP_CENTER,
+                className: 'toast_notification_cs'
```

Comment:
store position and className key and values in a seperate variable and destructure that variable here so that u dont need to change in pending, success and error in feature if u want to update it. u can just update in one place as all three pending, success and failure are using same key's and value
Note: Please add this in upcoming defactoring code changes so that